### PR TITLE
Fix spelling in std::pin documentation

### PR DESCRIPTION
**Pull request by AaronErhardt**
Recently I read the documentation of std::pin and found that it contains several spelling mistakes where pointer is misspelled "pointee". Actually, it might as well be correct for some reason because I've found that "pointee" appears in quite a few files of the Rust source code and has some use in LLVM (I wonder who had this incredibly bad idea to let a data type look like a spelling mistake). Also if it's correct this term comes out of nowhere and is never explained anywhere - at least as far I know. Not even searching the internet can tell me what a "pointee" should be.

I did hesitate a bit because if this was a mistake it should have been already fixed a long time ago but despite my research I couldn't find a single reason why "pointee" would be correct - at least inside the std::pin documentation.

TL;DR this PR replaces "pointee" with "pointer" in the doc for std::pin which might or might not be correct

### CODE REVIEW

**Comment by jonas-schievink** [comment]
"pointee" is correct. It describes the pointed-at object.

**Comment by jonas-schievink** [comment]
Also see https://en.wiktionary.org/wiki/pointee
